error LNK2019 when building the library NeoOnnx.dll

Hello.
I am building a library on windows 10, MSVC 2019 x64, protobuf 3.18.1 . I get two errors:

onnx.pb.obj : error LNK2019: ссылка на неразрешенный внешний символ "class google::protobuf::internal::ExplicitlyConstr
ucted<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > google::protobuf::inter
nal::fixed_address_empty_string" (?fixed_address_empty_string@internal@protobuf@google@@3V?$ExplicitlyConstructed@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@123@A) в функции "public: virtual class onnx::ValueInfoProto* __cdecl onnx::ValueInfoProto::New(void)const " (?New@ValueInfoProto@onnx@@UEBAPEAV12@XZ). [H:\libs\neoml2.0.22.0\Build\NeoOnnx\src\NeoOnnx.vcxproj]

onnx.pb.obj : error LNK2019: ссылка на неразрешенный внешний символ "struct std::atomic<bool> google::protobuf::interna
l::init_protobuf_defaults_state" (?init_protobuf_defaults_state@internal@protobuf@google@@3U?$atomic@_N@std@@A) в функции "public: virtual unsigned char * __cdecl onnx::TensorShapeProto_Dimension::_InternalSerialize(unsigned char *,class
google::protobuf::io::EpsCopyOutputStream *)const " (?_InternalSerialize@TensorShapeProto_Dimension@onnx@@UEBAPEAEPEAEPEAVEpsCopyOutputStream@io@protobuf@google@@@Z). [H:\libs\neoml2.0.22.0\Build\NeoOnnx\src\NeoOnnx.vcxproj]

H:\libs\neoml2.0.22.0\Build\NeoOnnx\src\Release\NeoOnnx.dll : fatal error LNK1120: неразрешенных внешних элементов: 2 [
H:\libs\neoml2.0.22.0\Build\NeoOnnx\src\NeoOnnx.vcxproj]

How fix it?
Thank you!

--- a/NeoOnnx/src/CMakeLists.txt
+++ b/NeoOnnx/src/CMakeLists.txt
@@ -1,3 +1,5 @@
+cmake_minimum_required(VERSION 3.11 FATAL_ERROR)
+
 if(NeoOnnx_BUILD_SHARED)
     add_library(${PROJECT_NAME} SHARED common.cpp)
 else()
@@ -99,8 +101,6 @@ if(DEFINED ENV{Protobuf_ROOT} AND (IOS OR ANDROID))
     list(APPEND CMAKE_FIND_ROOT_PATH $ENV{Protobuf_ROOT})
 endif()
 
-set(Protobuf_USE_STATIC_LIBS ON)
-
 find_package(Protobuf REQUIRED)
 
 if(NOT Protobuf_PROTOC_EXECUTABLE)
@@ -156,6 +156,15 @@ if(NeoOnnx_BUILD_SHARED AND NOT WIN32)
                 target_link_options(${PROJECT_NAME} PRIVATE -Wl,-exported_symbol,_${SYM})
             endforeach() 
     endif()
+endif()
+
+if(NOT Protobuf_LIBRARY)
+    message(FATAL_ERROR "Protobuf_LIBRARY not found!")
+endif()
+
+# Fix for a known protobuf Windows dll issue https://github.com/protocolbuffers/protobuf/issues/2502
+if((Protobuf_LIBRARY MATCHES ".*dll") OR (Protobuf_LIBRARY MATCHES ".*DLL"))
+    target_compile_definitions(${PROJECT_NAME} PRIVATE PROTOBUF_USE_DLLS)
 endif()
 
 target_link_libraries(${PROJECT_NAME} PRIVATE protobuf::libprotobuf NeoML)

--- a/NeoOnnx/src/CMakeLists.txt
+++ b/NeoOnnx/src/CMakeLists.txt
@@ -101,6 +101,8 @@ if(DEFINED ENV{Protobuf_ROOT} AND (IOS OR ANDROID))
     list(APPEND CMAKE_FIND_ROOT_PATH $ENV{Protobuf_ROOT})
 endif()
 
+set(Protobuf_USE_STATIC_LIBS ON)
+
 find_package(Protobuf REQUIRED)
 
 if(NOT Protobuf_PROTOC_EXECUTABLE)

